ext/dba/dba.c

```diff
@@ -1214,26 +1295,19 @@ PHP_FUNCTION(dba_handlers)
 /* {{{ List opened databases */
 PHP_FUNCTION(dba_list)
 {
-	zend_ulong numitems, i;
-	zend_resource *le;
-	dba_info *info;
-
 	if (zend_parse_parameters_none() == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	array_init(return_value);
 
-	numitems = zend_hash_next_free_element(&EG(regular_list));
-	for (i=1; i<numitems; i++) {
-		if ((le = zend_hash_index_find_ptr(&EG(regular_list), i)) == NULL) {
-			continue;
+	zval *zv;
+	ZEND_HASH_FOREACH_VAL(&DBA_G(connections), zv) {
```

Comment:
Because the keys in the connections HashTable are all strings, the underlying structure will be an actual hashmap.
So you can use `ZEND_HASH_MAP_FOREACH_VAL` for better looping performance and better code compaction.
The same holds for other loops over the connections able too.